### PR TITLE
Extract LocationsDirectory form object

### PR DESCRIPTION
**Pull request by dwhenry**
This extracts boolean flag issues out of the 
controller. This was required due to CSV filtering
not correctly set 'false' to false.

### CODE REVIEW

File: app/forms/locations_directory.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      @letter,
+      ALPHABETICAL_PAGINATE_CONFIG.dup,
+      &:title
+    )
+  end
+
+  private
+
+  def normalize_boolean_or_default(value, default)
+    return default if value.nil?
```
Comment (benlovell) [inline app/forms/locations_directory.rb]: `return default unless value` ?

File: spec/forms/locations_directory_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        it 'hidden location is not returned' do
+          expect(subject.locations).to be_empty
+        end
+      end
+
+      context 'when display_hidden is true' do
+        let(:params) { { locations_directory: { display_hidden: true } } }
+
+        it 'hidden location is returned' do
+          expect(subject.locations).to eq([location])
```
Comment (benlovell) [inline spec/forms/locations_directory_spec.rb]: Perhaps `contain_exactly(location)`? Your code is equivalent of course - but to my eyes `contain_exactly` reads more fluently.

File: spec/forms/locations_directory_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        it 'active location is not returned' do
+          expect(subject.locations).to be_empty
+        end
+      end
+
+      context 'when display_active is true' do
+        let(:params) { { locations_directory: { display_active: true } } }
+
+        it 'active location is returned' do
+          expect(subject.locations).to eq([location])
```
Comment (benlovell) [inline spec/forms/locations_directory_spec.rb]: Same here.

File: spec/forms/locations_directory_spec.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      expect(described_class.new(user, locations_directory: { display_hidden: 'true' }).display_hidden).to be_truthy
+      expect(described_class.new(user, locations_directory: { display_hidden: 'apples' }).display_hidden).to be_truthy
+    end
+  end
+
+  describe '#locations' do
+    subject { described_class.new(user, params) }
+
+    context 'when hidden location exists' do
+      let!(:location) { FactoryGirl.create(:location, hidden: true) }
```
Comment (benlovell) [inline spec/forms/locations_directory_spec.rb]: What did we standardise on with regards `FactoryGirl.create` or `create` bare?

File: app/forms/locations_directory.rb (hunk 1 above)
Comment (dwhenry) [inline app/forms/locations_directory.rb]: As discussed this differentiates between `nil` and `false`

File: spec/forms/locations_directory_spec.rb (hunk 2 above)
Comment (dwhenry) [inline spec/forms/locations_directory_spec.rb]: done

File: spec/forms/locations_directory_spec.rb (hunk 4 above)
Comment (dwhenry) [inline spec/forms/locations_directory_spec.rb]: Will do this in a separate PR as this project is currently setup to use `FactoryGirl.create`

**Comment by benlovell** [comment]
:shipit: 🚢
